src/contacts/mongodb-contacts.service.ts

```diff
@@ -50,9 +67,15 @@ export class MongoDbMContactsService implements ContactsService {
   }
   private convertToContact(contactDB: ContactDB): Contact {
     const { metaData, did, publicIdentity } = contactDB
+    let actualDid: IDidDocumentSigned
+    if (contactDB.toObject().signature && contactDB.toObject().did) {
+      actualDid = { ...did, signature: contactDB.toObject().signature }
+    } else {
+      actualDid = did
+    }
     return {
       metaData,
-      did,
+      did: actualDid,
```

Comment:
I guess 
```suggestion
      did: contactDB.signature && did ? {...did, signature: contactDB.signature} : did,
```
would also do it?

Reply:
in fact when you add the optional `{signature?: string}` to the signature you could even make this more concise by adding `signature` to 
```
const { metaData, did, publicIdentity } = contactDB
```